server/app.py

```
@@ -0,0 +1,189 @@
+        def on_game_over(reason: GameOverReason, data: Any):
+            nonlocal room
+
+            if reason == GameOverReason.WON:
+                log.info(f'player {data} won game {room}')
+                emit(events.GAME_OVER, {'reason': reason.value, 'winner': data.name}, to=room)
+            state.delete_all(room)
+
+        try:
+            game.play(player_id, card_id, on_game_over)
+        except ValueError as e:
+            log.error(e)
```

Ruff (0.13.2)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)